script/check-patch-diff.js

```diff
@@ -0,0 +1,40 @@
+const { execSync, spawnSync } = require('child_process');
+const path = require('path');
+const { inspect } = require('util');
+
+const srcPath = path.resolve(__dirname, '..', '..', '..');
+const patchExportPath = path.resolve(__dirname, 'export_all_patches.py');
+const configPath = path.resolve(__dirname, '..', 'patches', 'config.json');
+
+// Re-export all the patches to check if there were changes.
+execSync(`python ${patchExportPath} ${configPath}`, { cwd: srcPath });
+
+// Update indices.
+execSync('git update-index --refresh || true');
+
+// If the exit code is 0, then there were no differences.
+const { status } = spawnSync('git', ['diff-index', '--quiet', 'HEAD']);
+
+if (status !== 0) {
+  const committerName = execSync('git config user.name');
+  const committerEmail = execSync('git config user.email');
+
+  const env = Object.assign({}, process.env, {
+    GIT_COMMITTER_NAME: committerName || 'Electron Bot',
+    GIT_COMMITTER_EMAIL: committerEmail || 'anonymous@electronjs.org'
+  });
+
+  // Add updated patches.
+  execSync('git add patches');
+
+  // Commit updated patches.
+  const { status } = spawnSync('git', ['commit', '-m', 'chore: update patch indices'], {
+    env,
+    stdio: 'inherit'
+  });
+
+  if (status !== 0) {
+    console.error('Failed to update patches.');
+    process.exit(1);
+  }
```

Comment:
I think rather rely on the lint step to run `git add`, by analogy with the other linters we have that edit code files.